### ISSUE TITLE
Handling of user volume adjustment

### ISSUE THREAD

**Issue by meew0** [hydrabolt/discord.js#154]
As it currently stands, as of #135, voice playback volume defaults at 0.25 \* ffmpeg's default. If the user overwrites it, it will become the overwritten value (irrelevant of whether the set value is in decibels or as a factor) relative to **ffmpeg's default**. Now my problem with this behaviour is that making it relative to ffmpeg's default is stupid - we should be making it relative to **discord.js's default** of 0.25 * ffmpeg's default instead.

Reasoning:
* The current behaviour is confusing because if the user inputs 0dB, which *should* correspond to no change in gain relative to no volume setting at all, it will instead be louder (specifically, a 4x factor = +6 dB). This is unacceptable as decibels are a standardized unit.
* This same problem also applies to factor volumes, though this is less important - the user will just have to remember to make their specified factor volume relative to 0.25 if they want it to work like ffmpeg's factor volume where 1 is no change in volume.
* We are not writing ffmpeg here, we're writing discord.js, which abstracts from the internals - in my opinion, ffmpeg should just be an implementation detail that can possibly be changed at any time.
* Possibly most importantly, neither of these two scales actually mean anything, they're completely arbitrary. The most common reasoning *for* the current behaviour is that the user should expect discord.js not to mess with their set volume, completely ignoring the fact that even if discord.js didn't change it, Discord most certainly would. Consistency between local ffmpeg encoding/playback and Discord playback can, will and should never be achieved.

*TL/DR: User volumes should be relative to discord.js's internal scale, not ffmpeg's.*

**Comment by khionu** [comment]
As is, a person hears a volume. If they think it's too loud, they would think to lower it. By how much? People natively think in terms of what's in front of them. Ergo, I agree.

**Comment by TehSeph** [comment]
The fact that these values are arbitrary should be more than enough reason to leave it as is. If in the docs it is explained that the discord.js default value is `0.25` then it is understandable to the user that changing this value to `0.5` will double the default volume and not half. It doesn't make sense to scale these values in relation to the discord.js value because if the user is changing the value then they already understand that the value is being sent directly to ffmpeg in place of discord.js's value.

My vote is to leave it as it.

**Comment by khionu** [comment]
That assumes people are thinking about it more as FFMpeg's argument than Discord.JS's. I'm having to retrain myself to think of it as FFMpeg's for this discussion, but the majority of the users?

**Comment by meew0** [comment]
@TehSeph That doesn't apply to decibel volumes though - see my first point.

**Comment by TehSeph** [comment]
@khionu Hints the reasoning for stating in the docs that the default value is `0.25`. Then discord.js volume and ffmpeg volume become one in the same. No need for confusion.

@meew0 Decibels would still be in relation to ffmpeg's volume no matter what. Unless you want the library to include an entire equation for working out relative db adjustments... Which again is just more un-needed code. If the user is adjusting the volume then they should already be aware that the volume is in relation to ffmpeg's default and not the discord.js default.

**Comment by khionu** [comment]
Huh..... just thought of another point.... the user should already be reading up on FFMpeg for the purpose of file formats, this would be just one more little thing to check....

**Comment by meew0** [comment]
@TehSeph The specified decibel value shouldn't be in relation to anyone's default, it should be in relation to no change at all, i. e. not setting anything for the volume property. Relative dB adjustments are easy, just figure out the value for the default (10 log₁₀(0.25) ≈ -6 dB) and add the adjustment value onto that. In my opinion, volume adjustment should be intuitive enough that the user shouldn't necessarily have to read documentation for how it works, just what they have to do.

@khionu The user should not have to read up on ffmpeg's file formats..? I've never done it, it's easy enough if you understand that ffmpeg can highly likely handle your format just fine.

**Comment by meew0** [comment]
Some other points from Discord:
* It would be entirely possible to get rid of discord.js's scaling altogether and setting the default to 1, but this would make it too loud for some end users.
* The volume is already scaled relatively in a lot of other places (file data, file encoding, Discord, end user), getting rid of the relativity here makes it less consistent rather than more.
* A scaling where 1 / 0dB is the default is a lot more intuitive than one where 0.25 / -6dB is. The user should have to read and understand *how* to change the volume, but he shouldn't have to read and understand *what* to change it to if it can be avoided.

**Comment by Nerketur** [comment]
@meew0 Reading the issue, I partly agree with you.
As a *factor*, it should indeed be as stated, but as *dB* it should be absolute

Scaling the value, whatever it is, should make it `currval * scale`.  That makes sense, whether from the user standpoint or not.  However, if the user says "I want it to be exactly `x dB`", then it should be exactly that.

What the default value is (or is relevant to) doesn't make a single difference in the world, as long as the above holds.  As the current implementation seems to cause at least one of the above statements to become false,  I vote to change it.

**Comment by khionu** [comment]
Heh.... that's what I said, until I understood decibels better....

**Comment by hydrabolt** [comment]
I think we should leave it as `1` by default, and it's the end developer's responsibility to decide whether the audio they are playing is too loud or not. If they want to make sure that the audio they play is normal volume, they could just pass in `60dB` themselves. I do not think we should try to decide for the developer by default.

**Comment by meew0** [comment]
I agree, the best course of action would be not setting a default at all. And, just to clarify it here, 60dB is not normal volume if input into ffmpeg - as I said before, ffmpeg decibel input is relative to no change at all, so normal volume would be 0dB. (60dB would be a linear factor of 10^6 = a million.)

**Comment by macdja38** [comment]
I agree with leaving the value of one, to make adjusting it easier, though with the default value most of the developers can simply not pass it anything and be fine, it makes it more consistent. I do however think the doc's should very clearly state exactly how to lower the volume including a volume object and maybe an example line of code, just to make sure every developer can get their bot to a pleasant (or very painful) volume.

**Comment by khionu** [comment]
@hydrabolt it's already been adjusted to `.25`. 1 is painful to everyone, I have yet to hear of an exception.
